common/src/main/java/com/wynntils/mc/event/PacketEvent.java

```diff
@@ -17,6 +17,7 @@
 public class PacketEvent<T extends Packet<?>> extends GenericEvent<T> {
     private final T packet;
 
+    @SuppressWarnings("unchecked") // Technically an unsafe typecast, but nobody wants to pass up the packet class
```

Comment:
This is okay. There cracks where generics were glued into Java are showing up when dealing with `Class`es of generic types.